src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -774,6 +787,103 @@ private static CyclesAndDowngrades CreateState(List<GraphNode<RemoteResolveResul
             };
         }
 
+        public static AnalyzeResult<RemoteResolveResult> AnalyzeForCPVM(this GraphNode<RemoteResolveResult> root)
```

Comment:
Thinking from a NuGet SDK point of view:

Is this API needed? Can we modify the existing Analyze method instead?

Looking at the diff between your new CPVM versions of the methods, and the original version, I feel like the CPVM method could produce the same results as the non-CPVM method for non-CPVM projects, and we'll reduce business logic duplication.

If there are real differences between the CPVM and non-CPVM methods, can we do a detection in the `Analyze(...)` method and figure out which of the two private methods to call, so there's only 1 entry point? (public API doesn't change)

Reply:
Updated.